src/background/proxyHandler.js

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+import { Component } from "./component.js";
+import { Logger } from "./logger.js";
+import { Utils } from "./utils.js";
+ 
+const log = Logger.logger("ProxyHandler");
+
+class SiteContext {
+  constructor(context) {
+    context = context || {};
+    this.origin = context.origin;
+    this.cityName = context.cityName;
+    this.countryCode = context.countryCode;
+    this.excluded = context.excluded || false;
+    this.proxyInfo = context.proxyInfo;
+  }
+}
+ 
+/**
+ * This class manages tasks related to creating and storing 
+ * proxy information (contexts) for specific origins. 
+ */
+export class ProxyHandler extends Component {
```

Comment:
IMHO this class has `setSiteContexts` -> it should have a getter, currently every read need's to go through the Storage+JSON parser, which could be avoided.

Reply:
yep! Have a follow-up almost ready that does this and also sets siteContexts as a property to which the RequestListener, TabHandler, UI, etc can subscribe.